### ISSUE TITLE
Throw error when passed incorrect color mode instead of silently accepting it?

### ISSUE THREAD

**Issue by gregsadetsky** [processing/p5.js#5449]
### How would this new feature help [increase access](https://github.com/processing/p5.js/blob/main/contributor_docs/access.md) to p5.js?

p5.js could show an error message when `colorMode()` is passed an invalid color mode, instead of silently ignoring it.

#### Most appropriate sub-area of p5.js?

- [X] Friendly error system

#### New feature details:

`colorMode` accepts any string as its `mode`

I mistakenly passed `"HSL"` to it, e.g. a string, instead of the correct constant. I did not realize that `colorMode()` silently ignored my request to change the color mode, and stayed in RGB mode. It feels to me like this could be quite confusing to others as well.

I would suggest doing the following change: if the passed argument is not any of the known modes, throw an error, i.e.:

```javascript
p5.prototype.colorMode = function(mode, max1, max2, max3, maxA) {
  p5._validateParameters('colorMode', arguments);
  if (
    mode === constants.RGB ||
    mode === constants.HSB ||
    mode === constants.HSL
  ) {
  // ...
  // ...
  // *** New code below ***
  } else {
    throw new Error('invalid color mode')
  }
```

I'd be happy to do a commit/PR (with a test) for this. Thanks!

**Comment by willmartian** [comment]
It looks like an error is thrown when instantiating a new color, perhaps we could use the same message? https://github.com/processing/p5.js/blob/e46fa4d5ad9ba4cad771ac7b7bc01d1f099f3fa3/src/color/p5.Color.js#L41

Additionally, should we just convert the `mode` argument to lowercase before comparing? I feel like this might be an easy mistake to make.

**Comment by gregsadetsky** [comment]
I think that this error message is great, and your suggestion to convert to lowercase is really good as well! That would definitely have caught/saved me originally when passing "HSL".

**Comment by limzykenneth** [comment]
Being addressed in #7178 for 2.0
